### DISCUSSION TITLE
Queries API always return not null

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#15408] [Q&A] [answered]
I've created a SQL query, and trying to call it using the api.

`api/queries/queryName`

however this always returns 404. is there a setting I need to set?

https://docs.orchardcore.net/en/latest/docs/reference/modules/Queries/#apiqueriesname

I don't see any additional steps in the docs.

**Comment by MarGraz** [comment, marked as answer]
Hi @weirdyang, 

1. Did you had this error through the Queries preview in the _Admin panel > Search > Queries > All queries > Run_ (by clicking on your query)? 
2. Or did the error appear when you called `{your domain or localhost}/api/queries/queryname`?

**In the second scenario**, it's possible that you are using an **incorrect query name**, or you are **calling your API query anonymously**, from the browser or from Postman, without having permission to do so. It's usually better to invoke a query by providing OAuth 2.0 authentication. I experienced a similar issue [here](https://github.com/OrchardCMS/OrchardCore/issues/15381) (always try to search in the "Issues" section, not just in "Discussions", sometimes you can find helpful information). 
Essentially, to resolve it, you need to enable the "Open Id" features and configure it. More information are available [here](https://docs.orchardcore.net/fr/latest/docs/reference/modules/OpenId/). If you're familiar with Open Id, it's quite straightforward to set up.

Instead, **for a quick development testing**, you can expose your query API endpoint by changing the "Anonymous" role permission in the "Queries (OrchardCore.Queries)" section (from the admin panel: Roles > Anonymous > Edit), in this way everybody can call your query endpoint. However, this approach **_is not recommended_** in a production environment. Also for your microservices or website, activating OAuth 2.0 authentication is always the best practice. Keep in mind that the "Anonymous" user role has limited permissions, so depending on your needs, even in development, implementing OAuth 2.0 might be better than giving permission to every single module.

Additionally, **if you're a new user interested in contributing to the documentation** of Orchard Core, to enhance it, you can do so by editing the page directly on GitHub, in this specific case [here](https://github.com/OrchardCMS/OrchardCore/tree/main/src/docs/reference/modules/Queries) (you can always find the same page you mentioned, in a similar path in the OrchardCore solution "docs" folder).

I hope this helps, thank you.

**Reply by weirdyang** [reply]
appreciate it!

**Reply by MarGraz** [reply]
Happy that was helpful 😊
